packages/site/src/utils/storybook.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * This checks if the page is attempting to load a storybook path.
+ *
+ * If the url contains `?path`, we redirect to `/storybook/?path=___`.
+ *
+ * If the url is the root page, we redirect to `/storybook` unless the `?new`
+ * param is set, which forces the new docs site to render.
+ *
+ * Any other path is assumed to be owned by the new docs site, and will render.
+ */
+export function handleStorybookRedirect() {
```

Comment:
I adore this approach. I really like that you found a way to make this work within the confines of the docs site. And it really does work, I tried out the scenarios you provided and everything works as advertised.

I'm curious though, and we can chat about this: Could we potentially achieve the same effects with DNS/Rewrite rules at the domain/host level? If you and I were to sit down today and go through the backend UI and see what we can configure?

The ONLY reason, and I'll admit it's actually a relatively small reason and one I could be talked out of. You have to do the full round trip to the SPA, before you get redirected to Storybook. So technically the storybook experience gets a little worse while we roll out the new site.

Which ultimately, is not a huge deal.

So let's take a look today, and see if we can implement the logic you have here at that level. We could potentially leave a lot of this in place as a backup. And we'll time box ourselves so we don't burn too much trying to come up with a 'perfect' solution while the one you have here is proven and functional and it works and is easy to reason about.

Also either way, we will hold this PR back until we're okay with the new docs site leaking out if something doesn't work as intended in prod. Which I think is going to be early next week at the latest.